qiskit/circuit/gate.py

```diff
@@ -42,6 +47,28 @@ def to_matrix(self):
         """
         raise QiskitError("to_matrix not defined for this {}".format(type(self)))
 
+    def power(self, exponent):
+        if int(exponent) == exponent:
+            return super().power(exponent)
+
+        from qiskit.extensions.unitary import UnitaryGate  # pylint: disable=cyclic-import
+        # Should be diagonalized because it's a unitary.
+        decomposition, unitary = schur(self.to_matrix(), output='complex')
+        # Raise the diagonal entries to the specified power
+        decomposition_power = list()
+        for element in decomposition.diagonal():  # assert off-diagonal are 0
```

Comment:
This is a question for @eddieschoute

Reply:
If it's not, some bad numerical error must've happened, perhaps if the eigenvalues are close to zero for example. The resulting gate would not be valid so it definitely should be reported somehow.

Reply:
looks like this check should be done @1ucian0

Reply:
I will point out that all of this boils down to whether the input is a unitary or not. The non-diagonals being zero falls directly from this. The eigenvales or a unitary are necessarily nonzero, |lambda|=1 so that should be fine. It might make more sense to verify a gate is unitary up to some tolerance to begin with.

Reply:
I added the check (please, have a look if it makes sense or if it's inefficient) in cb1f4e8e

I could not trigger the check tho. Any example?